proposals/0448-type-variable-scoping.rst

```diff
@@ -414,7 +414,19 @@ Proposed Change Specification
    (Right now, pattern synonyms require all such quantifications to occur before any term arguments,
    but accepted proposal `#402`_ allows these quantifications to occur in any order in data constructors.)
 
-#. Any type variables mentioned in a type application are considered binding sites, shadowing any in-scope type variables.
+#. In accordance with the `Visibility Orthogonality Principle`_,
+   the rules that determine whether a variable occurrence is a binding site or a use site are not affected by the presence of a ``@``.
+   At the time of writing, the rules are as follows:
+
+   * Outside pattern signatures,
+     variable occurrences are considered binding sites,
+     shadowing any other in-scope variables.
+     It is an error to bring the same type variable into scope in two (or more) places within the same match group.
+
+   * Inside pattern signatures (i.e. on the right-hand side of ``pat :: sig``),
+     occurrences of in-scope type variables are usages,
+     whereas occurrences of out-of-scope type variables create implicit bindings.
+     It is allowed to mention the same out-of-scope variable more than once.
```

Comment:
This sentence doesn't have the intended payload, I think. The first such occurrence will bring the variable into scope, so the second occurrence is one of an out of scope variable.

Reply:
> The first such occurrence will bring the variable into scope

Not according to the definitions given here. The preceding sentence states quite clearly that occurrences of out-of-scope type variables create _implicit_ bindings. If the first occurrence brought the variable into scope, that would make it an explicit binding.

So I think the sentence says what I wanted it to say. Maybe I'm missing something.

Reply:
OK.